### PR TITLE
Upgrade project to use latest version of electron-vue

### PR DESCRIPTION
**Pull request by jack-chapman**
This PR addresses #92 . A dependency updated automatically by dependabot caused problems for fresh installs.

This change should hopefully prevent that kind of thing from happening again soon 👍 

Happy to make any changes!

### CODE REVIEW

**Comment by jack-chapman** [comment]
I can remove the package-lock.json if it's a problem!

**Comment by lauthieb** [comment]
Thanks @jack-chapman for this PR!
I will check that ASAP tomorrow :)

**Comment by jack-chapman** [comment]
^ Looks like the Travis build might have broken because the newer electron-vue version needs node 8.x.x

File: src/main/index.dev.js [hunk 1]
```diff
@@ -8,7 +8,7 @@
 /* eslint-disable */
 
 // Set environment for development
-process.env.NODE_ENV = 'development';
+// process.env.NODE_ENV = 'development'; THIS IS NOW SET WITH npm run dev
```
Comment (lauthieb) [inline src/main/index.dev.js]: Okay, nice :)

File: package.json [hunk 2]
```diff
@@ -74,77 +70,82 @@
     "nedb": "^1.8.0",
     "opencollective-postinstall": "^2.0.2",
     "source-sans-pro": "^2.45.0",
-    "vue": "^2.6.8",
-    "vue-clipboard2": "0.0.8",
+    "vue": "^2.5.16",
```
Comment (lauthieb) [inline package.json]: Why did you downgrade?

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.vue [hunk 3]
```diff
@@ -10,12 +8,14 @@ const noteNameCharacters = 'abcdef0123456789';
...
   let text = '';
-  for (let i = 0; i < 32; i += 1)
+  for (let i = 0; i < 32; i += 1) {
     text += noteNameCharacters.charAt(Math.floor(Math.random() * 16));
+  }
   return text;
 };
 
 export default {
+  template: require('./CreateNoteModal.html'),
```
Comment (lauthieb) [inline src/renderer/components/modals/create-note-modal/CreateNoteModal.vue]: is it the new way to declare template?

File: src/renderer/components/modals/settings-modal/SettingsModal.vue [hunk 4]
```diff
@@ -1,9 +1,8 @@
-<template src="./SettingsModal.html"> </template>
-
 <script>
 import { mapGetters, mapActions } from 'vuex';
 
 export default {
+  template: require('./SettingsModal.html'),
```
Comment (lauthieb) [inline src/renderer/components/modals/settings-modal/SettingsModal.vue]: same here. New way to declare template?

File: src/renderer/components/modals/update-note-modal/UpdateNoteModal.vue [hunk 5]
```diff
@@ -52,7 +51,7 @@ export default {
     updateNoteModal() {
       if (!this.containsDupFiles()) {
         if (this.gistsSelected) {
-          this.files.forEach(file => {
+          this.files.forEach((file) => {
```
Comment (lauthieb) [inline src/renderer/components/modals/update-note-modal/UpdateNoteModal.vue]: Hm. Why? :)

File: src/renderer/components/modals/update-note-modal/UpdateNoteModal.vue [hunk 6]
```diff
@@ -79,7 +78,7 @@ export default {
             }
           });
         } else {
-          this.files.forEach(file => {
+          this.files.forEach((file) => {
```
Comment (lauthieb) [inline src/renderer/components/modals/update-note-modal/UpdateNoteModal.vue]: same

File: src/renderer/components/modals/update-note-modal/UpdateNoteModal.vue [hunk 7]
```diff
@@ -117,7 +116,7 @@ export default {
       const map = new Map();
       let dupFiles = false;
 
-      this.files.forEach(file => {
+      this.files.forEach((file) => {
```
Comment (lauthieb) [inline src/renderer/components/modals/update-note-modal/UpdateNoteModal.vue]: same

File: src/renderer/components/notes-list/note-card/NoteCard.vue [hunk 8]
```diff
@@ -62,9 +61,9 @@ export default {
       this.$electron.shell.openExternal(link);
     },
     exportToCarbon(content) {
-      let url = `https://carbon.now.sh/?bg=rgba(0,0,0,0)&t=dracula&l=auto&ds=true&wc=true&wa=true&pv=43px&ph=57px&ln=false&code=`;
-      this.$electron.shell.openExternal(`${url}${encodeURI(content)}`)
-    }
+      const url = 'https://carbon.now.sh/?bg=rgba(0,0,0,0)&t=dracula&l=auto&ds=true&wc=true&wa=true&pv=43px&ph=57px&ln=false&code=';
```
Comment (lauthieb) [inline src/renderer/components/notes-list/note-card/NoteCard.vue]: NICE :D

**Comment by lauthieb** [comment]
@jack-chapman maybe you can change the .travis.yml file to fit with a node version 8?

**Comment by jack-chapman** [comment]
@lauthieb I'll fix the Travis build and put the Vue version back up to 2.6 later on today (I didn't notice that had changed when doing the electron-vue update.

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.vue (hunk 3 above)
Comment (jack-chapman) [reply src/renderer/components/modals/create-note-modal/CreateNoteModal.vue]: This was done to fix an error with the vue-loader. I can investigate an alternative fix, but this works.

**Comment by lauthieb** [comment]
Nice, thanks @jack-chapman !

**Comment by lauthieb** [comment]
OK for me, OK for Travis, let's merge :tada: @jack-chapman

**Comment by jack-chapman** [comment]
Hooray 🎉

**Comment by kodaman2** [comment]
Just FYI this branch broke the dev script.

npm run dev, before:
```
node .electron-vue/dev-runner.js
```

npm run dev, after:
```
NODE_ENV=development node .electron-vue/dev-runner.js
```

NODE_ENV is not recognized as an internal or external command.

**Comment by lauthieb** [comment]
Hi @kodaman2, so what we could do?

**Comment by jack-chapman** [comment]
@kodaman2 Have you run `npm install` to update your copy?

**Comment by lauthieb** [comment]
If it's a problem about Windows, maybe we could use [cross-env](https://www.npmjs.com/package/cross-env)

**Comment by kodaman2** [comment]
I was just trying to fix conflicts on my PR, and couldn't run dev, I just went back to the before, to be able to run `npm run dev`, not entirely sure why there is a NODE_ENV now, maybe @jack-chapman has a better idea or this was done automatically with the upgrade.

**Comment by kodaman2** [comment]
> @kodaman2 Have you run `npm install` to update your copy?

yes npm install was done, didn't have issues running dev before this merge btw, yes I am on windows. Is there some env variables I need to set now?

**Comment by lauthieb** [comment]
Okay, I already encounter this problem. Windows doesn't understand this type of ENV variable declaration. I will use cross-env and make a PR now.

**Comment by lauthieb** [comment]
image

**Comment by jack-chapman** [comment]
> Okay, I already encounter this problem. Windows doesn't understand this type of ENV variable declaration. I will use cross-env and make a PR now.

Apologies for this, I didn't realise windows does not recognise this type of ENV variable.

**Comment by lauthieb** [comment]
Here is my fix: #99, can you try?

**Comment by kodaman2** [comment]
> Here is my fix: #99, can you try?

Yes, fix worked! 😀

**Comment by lauthieb** [comment]
Thanks @kodaman2 for your answer.
Merged!

**Comment by lauthieb** [comment]
Here is the new release 1.2.3 of Code Notes : https://github.com/lauthieb/code-notes/releases/tag/1.2.3 :tada:
Congrats @kodaman2!

**Pull request merged**